userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/MainActivity.java

```diff
@@ -170,9 +177,32 @@ public void onAcquireTokenSilentClicked(final AcquireTokenFragment.RequestOption
 
         prepareRequestParameters(requestOptions);
 
-        callAcquireTokenSilent(requestOptions.getDataProfile().getText(),
-                getUserIdBasedOnUPN(requestOptions.getLoginHint(), requestOptions.getAuthority()),
-                requestOptions.getClientId().getText());
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                String uId = getUserIdBasedOnUPN(requestOptions.getLoginHint(), requestOptions.getAuthority());
+
+                if (TextUtils.isEmpty(uId)) {
+                    try {
+                        final UserInfo[] userInfos = mAuthContext.getBrokerUsers();
+
+                        for (UserInfo userInfo : userInfos) {
+                            if (userInfo.getDisplayableId().equalsIgnoreCase(requestOptions.getLoginHint())) {
+                                uId = userInfo.getUserId();
+                                break;
+                            }
+                        }
+                    } catch (OperationCanceledException | AuthenticatorException | IOException e) {
+                        Logger.e(TAG, "Unable to getbrokerusers from broker " + e.getMessage());
```

Comment:
Can we `Toast` this message?

If this fails and its not hooked up to a debugger, it'll just look like nothing happened

Reply:
Do we still want to try the silent call if `getBrokerUsers` fails like this? Seems like we should check for a non-error result before proceeding

Reply:
Only proceeding with silent if uId is non-null else showing a toast